rm-main/src/cli.rs

```diff
@@ -60,3 +63,56 @@ async fn add_torrent(config: &Config, torrent: String) -> Result<()> {
     };
     Ok(())
 }
+
+async fn fetch_rss(config: &Config, url: &str, filter: Option<&str>) -> Result<()> {
+    let mut transclient = transmission::utils::client_from_config(&config);
+    let content = reqwest::get(url).await?.bytes().await?;
+    let channel = rss::Channel::read_from(&content[..])?;
+    let re: Option<Regex> = {
+        if let Some(filter_str) = filter {
+            let res = Regex::new(&format!(r"{filter_str}"));
+            match res {
+                Err(e) => {
+                    eprintln!(
+                        "error constructing regex: {e}\nCheck if provided regex filter is valid."
+                    );
+                    std::process::exit(1);
+                }
+                Ok(re) => Some(re),
+            }
+        } else {
+            None
+        }
+    };
+    let urls = channel.items().iter().filter_map(|item| {
+        if let Some(title) = item.title() {
+            if let Some(re) = &re {
+                if re.is_match(title) {
+                    println!("{title} matches provided regex");
+                    return item.link();
+                } else {
+                    println!("{title} does not match provided regex");
+                }
+            } else {
+                return item.link();
+            }
+        }
+        None
+    });
+    for url in urls {
+        println!("downloading {url}");
```

Comment:
placeholder, not sure on what exactly to print. maybe include name?

Reply:
Yeah, maybe just the name would be better?